src/lib.rs

```diff
@@ -243,6 +246,25 @@ fn get_num_cpus() -> usize {
     cpus as usize
 }
 
+
+#[cfg(target_os = "macos")]
+fn get_num_physical_cpus() -> usize {
+    let mut cpus: i32 = 0;
+    let mut cpus_size = std::mem::size_of_val(&cpus);
+
+    let sysctl_name = CString::new("hw.physicalcpu").unwrap();
```

Comment:
I know. But it is simpler from a maintenance standpoin - and I just think it would be ugly.

There is also no need to optimize it into more unsafe code, because if someone is calling this in a hot loop, they are doing it wrong.